drivers/segger/note_sysview.c

```diff
@@ -275,7 +276,12 @@ static inline int sysview_isenabled_syscall(int nr)
  *
  ****************************************************************************/
 
-void sched_note_start(FAR struct tcb_s *tcb)
+void weak_function sched_note_start(FAR struct tcb_s *tcb)
+{
+  sysview_note_start(tcb);
+}
+
+void inline sysview_note_start(FAR struct tcb_s *tcb)
```

Comment:
will inline have any effect in such usage? I mean that `sysview_note_start` declared after `sched_note_start`.
Also, why do we need `sysview_note_start` at all?

Reply:
I  still think it cleaner to have a KCOFIG option for the sysview module to a prefix 
```
#if !defined(CONFIG_SYSTEMVIEW_PREFIX)
#  define sysview_prefix 
#else 
#  define sysview_prefix CONFIG_SYSTEMVIEW_PREFIX
// or define sysview_prefix  define sysview
#endif 

sysview_prefix_sched_note_resume(FAR struct tcb_s *tcb)
```

Reply:
Change it to use Kconfig.
@davids5 @pkarashchenko is cleaner/preferrable?

Reply:
@PetervdPerk-NXP Yes, I understood the idea, but need to refresh in my memory how `inline` works in C. I experimented with it long time ago and do not remember correctly. I mean do we need
```
void inline a(void)
{
}

void b(void)
{
  a();
}
```
or
```
void b(void)
{
  a();
}

void inline a(void)
{
}
```
I mean do we need to have `a` declared before `b` use it.
The `CONFIG_SYSTEMVIEW_PREFIX` seems to be cleaner, but most probably will require some `##` supplementary macros to be added to generate the names.